src/uproot/writing/_cascade.py

```diff
@@ -1760,7 +1761,7 @@ def add_directory(self, sink, name, initial_directory_bytes, uuid, flush=True):
             )
             next_key = subdirectory_key.copy_to(self._data.next_location)
         self._data.add_key(next_key, incremental=True)
-
+        next_key._location = self._data.next_location
```

Comment:
Finishing the above. (GitHub won't let me suggest changes through a deleted line.)

```suggestion

```